src/goto-instrument/opinionated_linter.cpp

```diff
@@ -0,0 +1,94 @@
+#include "opinionated_linter.h"
+#include "goto_program2code.h"
+
+#include <util/prefix.h>
+
+#include <iostream>
+
+int goto_hatert::score()
+{
+  int score = this->max_score();
+
+  // You only get two chances to ruin your program
+  const int decrement = this->max_score() / 2;
+
+  std::list<irep_idt> emp_l;
+  std::unordered_set<irep_idt> emp_s;
+  std::set<std::string> emp_ss;
+  for(auto &gf : this->gm.goto_functions.function_map)
```

Comment:
May be a `const` reference would also work?

`goto_program2code` shouldn't modify the go-to function